src/libsyntax/util/parser.rs

```diff
@@ -207,6 +207,28 @@ impl AssocOp {
             ObsoleteInPlace | Assign | AssignOp(_) | As | DotDot | DotDotEq | Colon => None
         }
     }
+
+    pub fn can_continue_expr_unambiguously(&self) -> bool {
```

Comment:
It'd be good to have a comment explaining what this function does here.

Reply:
done